sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,319 @@
+     * must be overridden to validate and support children that are wanted.
+     * </p>
+     *
+     * @param childConf a child of this component's Configuration
+     * @throws CheckstyleException if there is a configuration error.
+     * @see Configuration#getChildren
+     */
+
+    protected void setupChild(Configuration childConf)
+            throws CheckstyleException {
+        if (childConf != null) {
+            final String name = childConf.getName();
+            final String[] attributeNames = childConf.getAttributeNames();
+            for (String attributeName : attributeNames) {
+                if (!Arrays.stream(AllowedKeyInConfig.values()).anyMatch(
+                    item -> item.keyName.equals(attributeName))) {
+                    throw new CheckstyleException(String.format(
+                            "%s is not allowed an attribute in %s of %s",
+                            attributeName, name, ForbidCertainMethodsCheck.class.getSimpleName()));
+                }
+            }
+            String argCountRegex = null;
+            if (Arrays.stream(childConf.getAttributeNames()).anyMatch(
+                item -> item.equals(AllowedKeyInConfig.METHOD_ARG_COUNT.keyName))) {
+                argCountRegex = childConf.getAttribute(AllowedKeyInConfig.METHOD_ARG_COUNT.keyName);
+            }
+            rules.add(new Rule(name,
+                    childConf.getAttribute(AllowedKeyInConfig.METHOD_NAME.keyName),
+                    argCountRegex,
+                    childConf.getAttribute(AllowedKeyInConfig.REASON.keyName)));
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.METHOD_CALL,
+            TokenTypes.LITERAL_NEW,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL: {
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                if (dot.getType() == TokenTypes.IDENT) {
+                    final String methodName = dot.getText();
+                    final int numParams = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
+                }
+                // method that looks like: obj.method()
+                else {
+                    final String methodName = dot.getLastChild().getText();
+                    final int numParams = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
```

Comment:
Move `breaks` outside `if`/`else` to parent. Also move `checkForbiddenMethod`.
Declare `methodName` and `numParams` before `if`/`else`.
Example:
````
final int var;
if (true) {
  var = 1;
}
else {
  var = 0;
}
````